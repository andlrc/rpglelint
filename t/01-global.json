[
   {
      "column":7,
      "file":"t/01-global.rpgle",
      "line":2,
      "message":"global declaration 'global1' is not allowed",
      "severity":"warning"
   },
   {
      "column":8,
      "file":"t/01-global.rpgle",
      "line":3,
      "message":"global declaration 'global2' is not allowed",
      "severity":"warning"
   }
]
