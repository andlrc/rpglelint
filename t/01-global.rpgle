// rpglelint: -Wglobal
dcl-s global1 varchar(32);
dcl-ds global2;
  globalsub1 int(10);
  globalsub2 int(20);
end-ds;

dcl-proc a;
  dcl-s local1 varchar(32);
  dcl-ds local2;
    localsub1 int(10);
    localsub2 int(20);
  end-ds;
end-proc;
