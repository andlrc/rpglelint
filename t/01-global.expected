t/01-global.rpgle:2:7: warning: global declaration 'global1' is not allowed [-Wglobal]
 dcl-s global1 varchar(32);
       ^
t/01-global.rpgle:3:8: warning: global declaration 'global2' is not allowed [-Wglobal]
 dcl-ds global2;
        ^
